streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1302,6 +1302,19 @@ object ZSink extends ZSinkPlatformSpecificConstructors with Serializable {
         p(a).map(if (_) ((a :: as, true), Chunk.empty) else ((as, false), Chunk.single(a)))
     }.map(_._1.reverse)
 
+  /**
+   * Creates a sink which emits the number of elements processed
+   */
+  def count[A]: ZSink[Any, Nothing, Nothing, A, Long] = {
+    new SinkPure[Nothing, Nothing, A, Long] {
```

Comment:
To cut down on repetition, let's implement this with `Sink.foldLeft`?

Reply:
👍